idl/streaming.idl.hh

```diff
@@ -10,6 +10,8 @@
 #include "idl/token.idl.hh"
 #include "idl/uuid.idl.hh"
 
+#include "streaming/stream_fwd.hh"
```

Comment:
I don't understand how this include will prevent what the commit log claims it prevents.

Also, AFAIK we avoid adding includes to idl files.

Reply:
> I don't understand how this include will prevent what the commit log claims it prevents.

To use the serializer for streaming::plan_id, on needs to include the definition for `plan_id`.
Maybe it would have been better to describe this change as just making the idl header file self sufficient.

> Also, AFAIK we avoid adding includes to idl files.

Indeed in the past we used to include both the idl header and its c++ dependencies.
That changed in 1fda686f966ebe1ccb39c5c0e1e310184d80a3fa
and now we are able, and we should include the right dependencies from the idl.hh file
so that a c++ source file that include the generated idl header file for serializing / deserializing stuff
can just include that and it doesn't to to explicitly include also the respective dependencies.


As for preventing going out of sync, now that the idl header is self sufficient, if plan_id changes the serialization functions will fail to compile.

For example, applying this change:
```diff
diff --git a/streaming/stream_fwd.hh b/streaming/stream_fwd.hh
index b22970cff5..020a5abf65 100644
--- a/streaming/stream_fwd.hh
+++ b/streaming/stream_fwd.hh
@@ -18,6 +18,6 @@ class stream_result_future;
 class stream_session;
 class stream_state;
 
-using plan_id = utils::tagged_uuid<struct plan_id_tag>;
+using plan_id = utils::UUID;
 
 } // namespace streaming
```

results in:
```
$ ninja build/dev/message/messaging_service.o
In file included from message/messaging_service.cc:58:
build/dev/gen/idl/streaming.dist.hh:28:8: error: redefinition of 'serializer<utils::UUID>'
struct serializer<streaming::plan_id> {
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
build/dev/gen/idl/uuid.dist.hh:28:8: note: previous definition is here
struct serializer<utils::UUID> {
       ^
```